tests/phpunit/tests/option/wpDetermineOptionAutoloadValue.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/**
+ * @group option
+ *
+ * @covers ::wp_determine_option_autoload_value
+ */
+class Tests_WP_Determine_Option_Autoload_Value extends WP_UnitTestCase {
+	public function set_up() {
+		add_filter( 'wp_max_autoloaded_option_size', array( $this, 'filter_max_option_size' ) );
+		parent::set_up();
+	}
+
+	/**
+	 * @ticket 42441
+	 *
+	 * @dataProvider data_values
+	 *
+	 * @param $autoload
+	 * @param $expected
+	 */
+	public function test_determine_option_autoload_value( $autoload, $expected ) {
+		$test = wp_determine_option_autoload_value( null, '', '', $autoload );
+		$this->assertSame( $expected, $test );
+	}
+
+	public function data_values() {
+		return array(
+			'yes'   => array(
+				'autoload' => 'yes',
+				'expected' => 'on',
+			),
+			'true'  => array(
+				'autoload' => true,
+				'expected' => 'on',
+			),
+			'no'    => array(
+				'autoload' => 'no',
+				'expected' => 'off',
+			),
+			'false' => array(
+				'autoload' => false,
+				'expected' => 'off',
+			),
+			'null'  => array(
+				'autoload' => null,
+				'expected' => 'auto',
+			),
+		);
```

Comment:
See https://github.com/WordPress/wordpress-develop/pull/5671#discussion_r1522057356, this still needs to be updated.

For example:
```suggestion
		return array(
			'yes'      => array(
				'autoload' => 'yes',
				'expected' => 'on',
			),
			'on'       => array(
				'autoload' => 'on',
				'expected' => 'on',
			),
			'true'     => array(
				'autoload' => true,
				'expected' => 'on',
			),
			'no'       => array(
				'autoload' => 'no',
				'expected' => 'off',
			),
			'off'      => array(
				'autoload' => 'off',
				'expected' => 'off',
			),
			'false'    => array(
				'autoload' => false,
				'expected' => 'off',
			),
			'invalid'  => array(
				'autoload' => 'foo',
				'expected' => 'on',
			),
			'null'     => array(
				'autoload' => null,
				'expected' => 'auto',
			),
			'auto'     => array(
				'autoload' => 'auto',
				'expected' => 'auto',
			),
			'auto-on'  => array(
				'autoload' => 'auto-on',
				'expected' => 'auto',
			),
			'auto-off' => array(
				'autoload' => 'auto-off',
				'expected' => 'auto',
			),
		);
```